imports/ui/components/ShoppingListButton.jsx

```diff
@@ -1,18 +1,159 @@
+import "./ShoppingListButton.css";
 
 class ShoppingListButton extends Component {
   constructor(props) {
     super(props);
+    this.state = { left: false, checked: [] };
+  }
+
+  componentDidMount() {
+    this.props.getShoppingListItems();
   }
 
+  toggleDrawer = (side, open) => event => {
+    if (
+      event.type === "keydown" &&
+      (event.key === "Tab" || event.key === "Shift")
+    ) {
+      return;
+    }
+
+    this.setState({ ...this.state, [side]: open });
+  };
+
+  handleClick = () => {
+    let newShoppingList = {
+      createdBy: Meteor.userId(),
+      shoppingList: [],
+      createdAt: new Date()
+    };
+    this.props.addNewShoppngList(newShoppingList);
+    this.setState({ alreadyCreated: true });
+  };
+
+  handleToggle = value => () => {
+    const currentIndex = this.state.checked.indexOf(value);
+    const newChecked = [...this.state.checked];
+
+    if (currentIndex === -1) {
+      newChecked.push(value);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+
+    this.setState({ checked: newChecked });
+  };
+
+  sideList = (side, items) => (
+    <div
+      role="presentation"
+      onClick={this.toggleDrawer(side, false)}
+      onKeyDown={this.toggleDrawer(side, false)}
+      className="list-sector"
+    >
+      <Typography variant="h6" id="shopping-title">
+        Your personalized shopping list
+      </Typography>
+      <Divider />
+      <List>
+        {this.props.items.shoppingList.length === 0 ? (
+          <IconButton
+            style={{ display: this.state.alreadyCreated ? "none" : "" }}
+            onClick={this.handleClick}
+          >
+            Your shopping list is empty. Create a new shopping list by clicking
+            this area.
+          </IconButton>
+        ) : (
+          ""
+        )}
+        {items.map((text, index) => {
+          const labelId = `checkbox-list-secondary-label-${index}`;
+          return (
+            <ListItem key={index}>
+              <ListItemAvatar>
+                <Avatar
+                  alt="Remy Sharp"
+                  src={`/images/` + text.name + `.png`}
+                />
+              </ListItemAvatar>
+              <ListItemText
+                primary={text.name}
+                secondary={
+                  <React.Fragment>
+                    <Typography
+                      component="span"
+                      variant="body2"
+                      color="textPrimary"
+                    >
+                      {"$ " + text.price + " " + text.unit}
+                    </Typography>
+                    {" " + text.location.address}
+                  </React.Fragment>
+                }
+              />
+              <ListItemSecondaryAction>
+                <Checkbox
+                  edge="end"
+                  onChange={this.handleToggle(index)}
+                  checked={this.state.checked.indexOf(index) !== -1}
+                  inputProps={{ "aria-labelledby": labelId }}
```

Comment:
Do we want the checkboxed checked to persist after refresh?

Reply:
Do we want snackbar to stay open when we check boxes? Right now it collapses every time a box is checked.

Reply:
Checkbox now persist in database.

Reply:
> Do we want snackbar to stay open when we check boxes? Right now it collapses every time a box is checked.

I think you meant shoppinglist drawer right? I fixed that in the new commit where the drawer doesnt collapse when you check items.